.github/workflows/release.yml

```diff
@@ -0,0 +1,150 @@
+name: Release! Build & Package, Target Prod Install, NPM Publish (Tag)
+
+on: 
+  push: 
+    tags: 
+      # triggered only by major/minor/patch tags and those specifically tagged alpha. 
+      # standard prerelease tags do not trigger.
+      - 'v[0-9]+.[0-9]+.[0-9]+'
+      - 'v[0-9]+.[0-9]+.[0-9]+-alpha.*'
+
+  # TODO: implement dry-run ?
```

Comment:
I think either a dry-run option or the option to publish to staging instead would certainly be helpful :) although this is tricky with and automated trigger through push. 

I am wondering if you considered a mechanism where you do the release only through a manual workflow trigger, i.e., you can create the tag as part of this workflow, and the user would only provide the desired (version) tag as an input to the workflow? This way, a user would not need to know that pushing a tag will automatically release the artifacts and an unintentional release (by pushing a tag accidentally) would also be prevented.

Reply:
We can't really "publish to staging" per se. We do staging bucket on each pull request merge. We now publish to production bucket and then, if all is well, do the NPM publish which is technically independent. Versioning imposes a 'one time' success for both bucket and NPM. No overwrite is allowed. This behavior can be overridden but is highly discouraged. Which really leaves doing a dry run with minimal benefits (`--dry-run` flag is not really a full publish run, it stops before the publish step). Which is why it is left in the air...

What we do have is the (still) "secret" dev repo that works with two dev buckets and publishes a dev named NPM package using same code and workflows. There are some gotchas with deleting dev packages, but this is probably the best staging run we can get from such two-stage release process.

So - remove the TODO?

As for triggering via tag - comment below.

Reply:
Okay thanks for the explanation! From what I understand, it might indeed be the best choice not to implement a dry-run in this case. It might just unnecessarily complicate the workflow with a negligible benefit (if the dry-run code/ mechanics deviated too much from the actual publishing mechanics). I'd probably remove the dry-run TODO altogether then.

Reply:
Done.